packages/backend-defaults/src/entrypoints/rootHttpRouter/rootHttpRouterServiceFactory.ts

```diff
@@ -87,6 +87,10 @@ const rootHttpRouterServiceFactoryWithOptions = (
       const logger = rootLogger.child({ service: 'rootHttpRouter' });
       const app = express();
 
+      const trustProxy = config.getOptional<boolean | number>(
+        'backend.trustProxy',
+      );
```

Comment:
Not a super safe way to read this value, since it could end up being anything at runtime depending on what people put in config. Adding a full check for the type and then using the bool/number reader is a bit much to inline here though since they is intended as a piece of code that is easy to copy. A middleground would be if we could wrap this up somehow, but tbh I'm not seeing any great way to do that atm other than adding something new like `configureExpressApp(app, config)` but I don't love the idea of that.

We'll need to update https://backstage.io/docs/backend-system/core-services/root-http-router#via-code as well though

Reply:
I changed it to be just a boolean for now. I think the `Config` requires some way to get either one without copy-pasting always the same type check code.